### ISSUE TITLE
ISSUE-3: add testkey to opendkim.eom()

### ISSUE THREAD

**Issue by godsflaw** [godsflaw/node-opendkim#3]
Add the `testkey` argument to `opendkim.eom()`.

- [ ] code
- [ ] tests
- [ ] documentation

for more information see:
http://www.opendkim.org/libopendkim/dkim_eom.html

Bounty:
https://ethlance.com/#/job/240

**Comment by desttinghim** [comment]
The testkey argument is for a pointer to a bool, which will be set to true if the key found is a test key. Obviously we don't want to do the exact same thing here. I think returning an object with two fields would probably be appropriate:
```
{
    dkim: DKIM_STAT_OK,
    testkey: false
}
```
Thoughts?
EDIT: The current code seems to return the `testkey` value as an indicator of success. Is this a bug?
EDIT2: Should a string be returned for the `dkim` field or should we just pass the error number through?

**Comment by godsflaw** [comment]
@desttinghim hmm, setting aside `testkey` for a second, the status of the call is already handled.  If it does not throw an error, it was a success.  This is consistent with other calls.

Now on to `testkey`.  It does look like I added the ability to toss this value back to the caller regardless if they asked for it or not.  I'm not sure I like how I handled this though.  Part of the problem is a translation from the JS calling style, through C++, and to the C calling style.  I feel like we should take pains to perhaps make the JS side "feel" like JS.  If there is a bug, it's that I didn't allow this to method chain like,
```
try {
  var opendkim = new OpenDKIM();
  var domain = opendkim
    .verify({
      id: undefined
    })
    .header({
      header: header,
      length: header.length
    })
    .eoh()
    .body({
      body: body,
      length: body.length
    })
    .eom()
    .sig_getdomain();
} catch (err) {
  console.log(err);
}
```
Now, that being said, I am currently rewriting a bunch of the calls into this library to use promises so they can `async`/`await`.  You can see early work here: https://github.com/godsflaw/node-opendkim/pull/55

AFAIK, using `async`/`await` in this way doesn't really allow for method chaining, unless you count how it's basically promise chaining under the hood.  In fact, one does not need to use `async`/`await`, and can in theory just chain promise calls together.  Right now, I was hoping to just make blocking calls into `libopendkim` into promises so they can run async in a worker thread.  I didn't want to convert everything because of the volume of that work.

Back to this question.  I think what we want, is a way to signal to `opendkdim.eom()` that we are interested in the return value of `testkey`.  Perhaps `opendkim.eom({ testkey : true })` when calling.  If true, it returns what it does now, if not true, it returns `this` (the instantiated OpenDKIM object).  What do you think?

**Comment by desttinghim** [comment]
Sounds good to me.

**Comment by desttinghim** [comment]
I've spent some time working on this, and it seems like this is almost working. Only problem I'm running into is actually with returning `info.This()` from the function. An object gets returned, but there's nothing in it. Calling `opendkim.eom().sig_getdomain();` gives me this error:
```
TypeError: opendkim.header(...).eoh(...).body(...).eom(...).sig_getdomain is not a function
```
[Here's a link to the code.](https://github.com/godsflaw/node-opendkim/compare/dev...desttinghim:ISSUE-3)

EDIT: Looks like I was using master instead of dev. Somehow that was what was breaking it?
EDIT2: The branch at the link has been updated
